rspace/src/main/scala/coop/rchain/rspace/history/package.scala

```diff
@@ -81,8 +84,8 @@ package object history {
     }
 
   // TODO(ht): make this more efficient
-  private[this] def commonPrefix[A](a: Seq[A], b: Seq[A]): Seq[A] =
-    a.zip(b).takeWhile { case (l, r) => l == r }.map(_._1)
+  private[this] def commonPrefix[A: Eq](a: Seq[A], b: Seq[A]): Seq[A] =
+    a.zip(b).takeWhile { case (l, r) => l === r }.map(_._1)
```

Comment:
Thanks for that.  Any thoughts on `Stream` vs `Iterator`?

Reply:
Not really. The only thing that comes into my mind is memory allocated for each of them. Streams are scala-ish so personally I would go with them but just because it's native scala.